File: Features/Tratamientos/Tratamientos.Domain/Models/Tratamiento.cs
namespace feedback_zoologic.Features.Tratamientos.Domain
{
    public class Tratamiento : Entity, IAggregateRoot
    {
        public Tratamiento
        (
            int tratamientoId,
            string nombreTratamiento,
            int especieId,
            string zonaId,
            DateTime fechaEntrada,
            DateTime fechaSalida,
            int usuarioId,
            string razon,
            string procedencia
        )
        {
            TratamientoId = tratamientoId;
            NombreTratamiento = nombreTratamiento;
            EspecieId = especieId;
            ZonaId = zonaId;
            FechaEntrada = fechaEntrada;
            FechaSalida = fechaSalida;
            UsuarioId = usuarioId;
            Razon = razon;
            Procedencia = procedencia;
        }
        
        public int TratamientoId { get; set; }
        public string NombreTratamiento { get; set; } = null!;
        public int EspecieId { get; set; }
        public string NombreComun { get; set; } = null!;
        public string ZonaId { get; set; } = null!;
        public string NombreZona { get; set; } = null!;
        public DateTime FechaEntrada { get; set; }
        public DateTime FechaSalida { get; set; }
        public int UsuarioId { get; set; }
        public string NombreUsuario { get; set;} = null!;
        public string Razon { get; set; } = null!;
        public string Procedencia {get ; set;} = null!;
        
        public Tratamiento UpdateOrganizacionId(int tratamientoId)
        {
            if(tratamientoId != 0)
                TratamientoId = tratamientoId;
            return this;
        }
    }
}

File: ClientApp/interfaces/tratamiento.ts
export interface ITratamiento {
    TratamientoId: number;
    NombreTratamiento: string;
}

export interface ITratamientoCurrent {
    TratamientoId: number;
    NombreTratamiento: string;
}

export interface ITratamientoCreate {
    NombreTratamiento: string;
}

export interface ITratamientoUpdate {
    TratamientoId: number;
    NombreTratamiento: string;
}

export interface ITratamientoAplicado {
    TratamientoAplicadoId: number;
    TratamientoId: number;
    NombreTratamiento: string;
    AnimalId: number;
    IdentificadorUnico: string
    HabitatId: number;
    Nombre: string;
    FechaEntrada: Date | string;
    FechaSalida: Date | string;
    UsuarioId: number;
    NombreUsuario: string;
    Razon: string;
}

export interface ITratamientoAplicadoCreate {
    TratamientoId: number;
    AnimalId: number;
    HabitatId: number;
    FechaEntrada: Date | null;
    FechaSalida: Date | null;
    UsuarioId: number;
    Razon: string;
    ActivarFechas?: boolean;
}

export interface ITratamientoEspecie {
    TratamientoEspecieId: number;
    TratamientoId: number;
    NombreTratamiento: string;
    EspecieId: number;
    NombreComun: string;
}

export interface ITratamientoEspecieCurrent {
    TratamientoEspecieId: number;
    TratamientoId: number;
    NombreTratamiento: string;
    EspecieId: number;
    NombreComun: string;
}

export interface ITratamientoEspecieCreate {
    TratamientoId: number;
    EspecieId: number;
}

export interface ITratamientoEspecieUpdate {
    TratamientoEspecieId: number;
    TratamientoId: number;
    EspecieId: number;
}

